File: View/WindowManager/FloatView/src/main/java/com/view/wm/IApplication.java
package com.view.wm;

import com.yline.application.BaseApplication;

/**
 * App 入口
 *
 * @author yline 2018/1/11 -- 11:54
 * @version 1.0.0
 */
public class IApplication extends BaseApplication {

    @Override
    public void onCreate() {
        super.onCreate();

        FloatService.launcher(this);
    }

    @Override
    public void onTerminate() {
        // 防止内存泄漏
        FloatService.unRegisterCircleClickListener();

        super.onTerminate();
    }
}

File: System/AppOther/Manager/src/main/java/com/manager/IApplication.kt
package com.manager

import com.yline.application.BaseApplication
import java.util.concurrent.Callable
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors
import java.util.concurrent.Future

/**
 * 入口
 * @author yline 2018/6/5 -- 13:45
 * @version 1.0.0
 */
class IApplication : BaseApplication() {
    companion object {
        private var sFixedThreadPool: ExecutorService? = null

        fun fixedThreadExecutor(runnable: Runnable) {
            if (null == sFixedThreadPool) {
                sFixedThreadPool = Executors.newFixedThreadPool(5)
            }
            sFixedThreadPool?.execute(runnable)
        }

        fun <T> fixedThreadExecutor(callable: Callable<T>): Future<T> {
            if (null == sFixedThreadPool) {
                sFixedThreadPool = Executors.newFixedThreadPool(5)
            }
            return sFixedThreadPool!!.submit(callable)
        }
    }

    override fun onCreate() {
        super.onCreate()
    }
}